internal/persistence/sql/migrations/migratest/migration_test.go

```diff
@@ -39,44 +108,76 @@ func TestMigrations(t *testing.T) {
 
 			var c *pop.Connection
 			var err error
+			c, err = pop.NewConnection(&pop.ConnectionDetails{URL: db.Conn})
+			require.NoError(t, err)
+			require.NoError(t, c.Open())
+			t.Cleanup(func() { c.Close() })
 			for i := 0; i < 120; i++ {
-				c, err = pop.NewConnection(&pop.ConnectionDetails{URL: db.Conn})
-				require.NoError(t, err)
-				require.NoError(t, c.Open())
 				if err := c.Store.(interface{ Ping() error }).Ping(); err == nil {
 					break
 				}
 				time.Sleep(time.Second)
 			}
 			require.NoError(t, c.Store.(interface{ Ping() error }).Ping())
 
-			tm := popx.NewTestMigrator(t, c, fsx.Merge(networkx.Migrations, os.DirFS("../sql")), os.DirFS("./testdata"), l)
+			tm, err := popx.NewMigrationBox(
+				fsx.Merge(sql.Migrations, networkx.Migrations, os.DirFS("./testdata")),
+				popx.NewMigrator(c, l, nil, 1*time.Minute),
+				popx.WithGoMigrations(uuidmapping.Migrations),
+				withTestdata(os.DirFS("./testdata")),
+			)
```

Comment:
Yes, the first testdata can go.

What do you mean by single source of truth? The migration box is assembled differently in the registry vs. in this test case because of the testdata that needs to be loaded.